presto-spi/src/main/java/com/facebook/presto/spi/plan/ConnectorJoinNode.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.plan;
+
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorJoinNode
+        extends PlanNode
+{
+    private final List<PlanNode> sources;
+    private final JoinType type;
+    private final Set<EquiJoinClause> criteria;
+    private final Set<RowExpression> filters;
+    private final List<VariableReferenceExpression> outputVariables;
+
+    public ConnectorJoinNode(
+            PlanNodeId id,
+            List<PlanNode> sources,
+            Optional<PlanNode> statsEquivalentPlanNode,
+            JoinType type,
+            Set<EquiJoinClause> criteria,
+            Set<RowExpression> filters,
+            List<VariableReferenceExpression> outputVariables)
+    {
+        super(Optional.empty(), id, statsEquivalentPlanNode);
+        this.sources = requireNonNull(sources, "sources is null");
+        this.type = requireNonNull(type, "type is null");
+        this.criteria = requireNonNull(criteria, "criteria is null");
+        this.filters = requireNonNull(filters, "filters is null");
+        this.outputVariables = requireNonNull(outputVariables, "outputVariables is null");
+    }
+
+    @Override
+    @JsonProperty
```

Comment:
Is this necessary? 
If Json serialization is necessary, please add a JsonCreator construction method, if not remove these 
JsonProperty annotations.

Similar for other fields

Reply:
Removed the json attributes.